src/UserSwitch/UserSwitch.tsx

```diff
@@ -0,0 +1,278 @@
+import React, {
+  FC,
+  MouseEvent,
+  MouseEventHandler,
+  ReactElement,
+  useState,
+} from 'react';
+import {
+  isPseudonymizedMember,
+  isSessionExpired,
+  isError,
+} from '@graasp/utils';
+import { makeStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+import Menu from '@material-ui/core/Menu';
+import Tooltip from '@material-ui/core/Tooltip';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import Box from '@material-ui/core/Box';
+import { Divider } from '@material-ui/core';
+import { Map } from 'immutable';
+import MenuItem from '@material-ui/core/MenuItem';
+import {
+  HEADER_USERNAME_MAX_WIDTH,
+  SHORT_TEXT_WIDTH,
+  SMALL_AVATAR_SIZE,
+} from '../constants';
+import { ImmutableMember, Member, Variant } from '../types';
+import Avatar from '../Avatar';
+import Button from '../Button';
+import { Skeleton } from '@material-ui/lab';
+
+const useStyles = makeStyles((theme) => ({
+  wrapper: {
+    display: 'flex',
+    alignItems: 'center',
+    '&:hover': {
+      cursor: 'pointer',
+    },
+  },
+  mainUsername: {
+    margin: theme.spacing(0, 2),
+    maxWidth: HEADER_USERNAME_MAX_WIDTH,
+  },
+  mainAvatar: {
+    width: 70,
+    height: 70,
+    marginRight: theme.spacing(2),
+  },
+  profileButton: {
+    margin: 0,
+    marginTop: theme.spacing(1),
+  },
+  avatar: {
+    width: 30,
+    height: 30,
+  },
+  skeletonUsername: {
+    margin: theme.spacing(0, 2),
+  },
+}));
+
+interface Props {
+  useAvatar: Function;
+  member?: ImmutableMember;
+  members?: Member[];
+  onSeeProfileClick?: MouseEventHandler;
+  onMemberClick?: (_id: string) => MouseEventHandler;
+  seeProfileText?: string;
+  signedOutTooltipText?: string;
+  Actions?: JSX.Element | JSX.Element[];
+  ButtonContent?: JSX.Element;
+  isMemberLoading?: boolean;
+  buttonId?: string;
+  buildMemberMenuItemId?: (id: string) => string;
+  seeProfileButtonId?: string;
+}
+
+const UserSwitch: FC<Props> = ({
+  useAvatar,
+  onSeeProfileClick,
+  ButtonContent,
+  Actions,
+  buttonId,
+  seeProfileButtonId,
+  buildMemberMenuItemId,
+  member = Map(),
+  members = [],
+  isMemberLoading = false,
+  onMemberClick = () => () => undefined,
+  seeProfileText = 'See Profile',
+  signedOutTooltipText = 'You are not signed in.',
+}) => {
+  const classes = useStyles();
+  const [anchorEl, setAnchorEl] = useState<(EventTarget & Element) | null>(
```

Comment:
Consider preferring [`useRef`](https://reactjs.org/docs/hooks-reference.html#useref) to `useState` if applicable? If I understood correctly, you only use it to locate the root `Box`. In which case you probably won't need `handleClick` anymore. I would find it more readable if you track the open state separately as a boolean in `const [isOpen, setOpen] = useState(false)` for instance. Let me know if this is not possible.

Reply:
Nice comment, but there's some issues here:
- This is actually a copy paste of mui's implementation of the menu. It doesn't change in the new version, so I guess it's safe to use it like this.
- An implementation with ref would work. But since the button can be custom, I cannot really set the ref on it. The wrapper component is a Box and I cannot set ref to it.
So I think I'll let it like that.